### PR TITLE
redis: update Redis images for persistence optimizations

### PR DESCRIPTION
**Pull request by tsenart**
This PR is a follow up to https://github.com/sourcegraph/sourcegraph/pull/8114

### CODE REVIEW

**Comment by beyang** [comment]
Not sure about removing the redis-cache volume argument. Is the intention here to use the container FS, rather than the host FS? @slimsag should review.

**Comment by tsenart** [comment]
@slimsag: Why did you push this separate change? https://github.com/sourcegraph/deploy-sourcegraph-docker/commit/02c695926587e1ce9c01481ba675abc45d309e8e

@beyang: The idea is that red is cache is ephemeral so there’s no need in persisting anything to disk.

File: docker-compose/docker-compose.yaml [hunk 1]
```diff
@@ -532,7 +532,7 @@ services:
   #
   redis-cache:
     container_name: redis-cache
-    image: index.docker.io/sourcegraph/redis-cache:19-04-16_6891de82@sha256:4cbfac8af0abb673899250d4fd859cc477d6426de519e9deb71e454e18322499
+    image: index.docker.io/sourcegraph/redis-cache:20-01-30_c903717e@sha256:c069a4589420bc6d18a7d09ebaf5416fa6407666770cf650566dfd450bba65cf
```
Comment (slimsag) [inline docker-compose/docker-compose.yaml]:
```suggestion
    image: 'index.docker.io/sourcegraph/redis-cache:20-01-30_c903717e@sha256:c069a4589420bc6d18a7d09ebaf5416fa6407666770cf650566dfd450bba65cf'
```

File: docker-compose/docker-compose.yaml [hunk 2]
```diff
@@ -546,7 +546,7 @@ services:
   #
   redis-store:
     container_name: redis-store
-    image: 'index.docker.io/sourcegraph/redis-store:19-04-16_6891de821@sha256:56426d601ce1f6d63088fea1cefa61f69a2e809c7d90fc1d157cca63cf81b277'
+    image: index.docker.io/sourcegraph/redis-store:21-01-30_c903717e@sha256:e8467a8279832207559bdfbc4a89b68916ecd5b44ab5cf7620c995461c005168
```
Comment (slimsag) [inline docker-compose/docker-compose.yaml]:
```suggestion
    image: 'index.docker.io/sourcegraph/redis-store:21-01-30_c903717e@sha256:e8467a8279832207559bdfbc4a89b68916ecd5b44ab5cf7620c995461c005168'
```

**Comment by slimsag** [comment]
@tsenart Sorry, I only saw https://github.com/sourcegraph/deploy-sourcegraph/pull/515 and it looked unresolved so I fixed it here. I should've checked the PRs on this repo first, my bad.

I fixed the merge conflicts and will merge this, LGTM

**Pull request merged**